templates/scripts/pre-env.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+set -e
+
+## Sendgrid
+PROJECT_NAME=<% .Name %>
+RANDOM_SEED=<% index .Params `randomSeed` %>
+REGION=<% index .Params `region` %>
+SENDGRID_ENDPOINT=https://api.sendgrid.com/v3/
+SENDGRID_API_KEY=$(aws secretsmanager get-secret-value --region=${REGION} --secret-id=${PROJECT_NAME}-sendgrid-${RANDOM_SEED} --query "SecretString" | sed "s/\"//g")
+PROJECT_DIR=..
+DOMAIN_PREFIX=mail.
+
+if [ "${ENVIRONMENT}" == "stage" ]; then
+  SENDGRID_DOMAIN="${DOMAIN_PREFIX}<% index .Params `stagingHostRoot` %>"
+elif [ "${ENVIRONMENT}" == "prod" ]; then
+  SENDGRID_DOMAIN="${DOMAIN_PREFIX}<% index .Params `productionHostRoot` %>"
+fi
+
+if [ ! -z "${SENDGRID_API_KEY}" ]
+then
+  MATCHING_DOMAIN_COUNT=$(curl -XGET "${SENDGRID_ENDPOINT}whitelabel/domains" -H "Authorization:Bearer ${SENDGRID_API_KEY}" | jq -c "[ .[] | select(.domain == \"${SENDGRID_DOMAIN}\")] | length");
```

Comment:
All this stuff is quite vulnerable to them making changes to their API, but it seems like it's the only option.. :feelsbadman:

Reply:
Yeah hopefully they won't make breaking changes on their `/v3` api, then we would have the option to manually upgrade it 🤞